crates/ruff/src/args.rs

```diff
@@ -515,101 +554,184 @@ impl From<&LogLevelArgs> for LogLevel {
     }
 }
 
+/// Struct to encapsulate all configuration-related arguments passed via the CLI.
+///
+/// Three kinds of configuration arguments can be provided via the CLI;
+/// each kind is represented by a separate field on this struct.
+#[derive(Default)]
+pub struct ConfigArguments {
+    /// Path to a pyproject.toml or ruff.toml config file (etc.).
+    /// Either 0 or 1 config file paths may be provided on the command line.
```

Comment:
Just wondering, will we allow more in the future?

Reply:
We could do; I think it would be feasible implementation-wise, and I don't think it would be particularly _confusing_ for users to have that behaviour. Has it been asked for by users, do you know?

Reply:
I'm just thinking about the cargo implementation. I haven't heard an ask for it but I wouldn't be surprised! We can punt on it for now though we might want to be careful about precedence (i.e. as discussed in https://github.com/astral-sh/ruff/pull/9599#discussion_r1480577993) to avoid having it be a breaking change in the future.

Reply:
Cool -- let's punt on it for now. Very much open to this if it is asked for, though!